cloud/services/object_storage_buckets.go

```diff
@@ -109,21 +96,13 @@ func createObjectStorageKey(ctx context.Context, bScope *scope.ObjectStorageBuck
 	return key, nil
 }
 
-func RevokeObjectStorageKeys(ctx context.Context, bScope *scope.ObjectStorageBucketScope, secret *corev1.Secret) error {
-	if secret == nil {
-		return errors.New("unable to read access keys from nil secret")
-	}
-
-	keyIDs, err := bScope.GetAccessKeysFromSecret(ctx, secret)
-	if err != nil {
-		bScope.Logger.Error(err, "Failed to read secret with access keys to revoke; must be manually revoked")
-
-		return fmt.Errorf("failed to read secret %s with access keys: %w", secret.Name, err)
-	}
-
+func RevokeObjectStorageKeys(ctx context.Context, bScope *scope.ObjectStorageBucketScope) error {
 	var errs []error
-	for _, keyID := range keyIDs {
-		if err := revokeObjectStorageKey(ctx, bScope, keyID); err != nil {
+	for _, keyID := range []*int{bScope.Bucket.Status.ReadWriteKeyID, bScope.Bucket.Status.ReadOnlyKeyID} {
```

Comment:
using a for loop here seems a bit odd unless we were iterating over something like `Status.AccessKeys`. Will there only ever a single key for each rather than a list for both RW and RO? Also do we need to distinguish between RW vs RO?

Reply:
There should only ever be a single key for each. I think storing the IDs as an array makes sense. Will make that change.